Content.Server/Atmos/Piping/Other/EntitySystems/GasMinerSystem.cs

```diff
@@ -25,10 +25,17 @@ private void OnMinerUpdated(Entity<GasMinerComponent> ent, ref AtmosDeviceUpdate
         {
             var miner = ent.Comp;
 
+            if(!GetValidEnvironment(ent, out var environment))
+            {
+                miner.Idle = false;
+                return;
+            }
+
             // SpawnAmount is declared in mol/s so to get the amount of gas we hope to mine, we have to multiply this by
             // how long we have been waiting to spawn it and further cap the number according to the miner's state.
-            var toSpawn = CapSpawnAmount(ent, miner.SpawnAmount * args.dt, out var environment);
-            if (toSpawn <= 0f || environment == null || !miner.Enabled || !miner.SpawnGas.HasValue)
+            var toSpawn = CapSpawnAmount(ent, miner.SpawnAmount * args.dt, environment);
+            miner.Idle = toSpawn == 0;
```

Comment:
Could you double check C#'s const equality rules? I'm not sure if the literal `0f == 0` since `0` is technically a double. Or, did you test that this check actually works in game?

Reply:
It will be converted down to a float and compared without issue. Unsure if this happens statically at compile time or at runtime however.

I've tested it a few times with the content and it works as expected.

Reply:
[looks like the compiler catches it](https://sharplab.io/#v2:EYLgHgbALANALiAlgGwD4AEBMBGAsAKHQGYACLEgYRIG8CT6zTgB7Z5EgM1YAoBKGugyEdkzAIZwSALwCmAJ2YcSAXhIAGAHRqOAbkFD6LNiTkyAzgFdkcMSunzFK1Wr34DDEeMmyFAEzsu+gZG7KaW1sB2Psz+ys6u7vToAOwm5lY2CQwAvgTZQA===)

Reply:
Thanks for checking!